lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3018,7 +3018,7 @@ protected function _storeAssociationMapping(array $assocMapping)
      * Registers a custom repository class for the entity class.
      *
      * @param string|null $repositoryClassName The class name of the custom mapper.
-     * @psalm-param class-string<EntityRepository>|null $repositoryClassName
+     * @psalm-param class-string<ObjectRepository>|null $repositoryClassName
```

Comment:
:thinking: what makes you widen this type?

Reply:
Custom repositories don't have to extend `EntityRepository`. Is is sufficient if they implement `ObjectRepository`.

https://github.com/doctrine/orm/blob/193c3abf0e5f7a7e2699e7c7ac37c1e713d79ef8/lib/Doctrine/ORM/Exception/InvalidEntityRepository.php#L14

https://github.com/doctrine/orm/blob/193c3abf0e5f7a7e2699e7c7ac37c1e713d79ef8/lib/Doctrine/ORM/Configuration.php#L832

Reply:
Hmm, but this also means that the types on the entity manager are wrong. I'll remove the widening for now and try to straighten this out in a separate PR.

Reply:
Moved to #9531.